hnn_core/network.py

```diff
@@ -311,55 +322,75 @@ def _create_all_src(self):
-                if type in ('L2_pyramidal', 'L5_pyramidal', 'L2_basket',
-                            'L5_basket'):
-                    Cell = type2class[type]
-                    if type in ('L2_pyramidal', 'L5_pyramidal'):
-                        self.cells.append(Cell(gid, pos, self.params))
-                    else:
-                        self.cells.append(Cell(gid, pos))
-                    pc.cell(
-                        gid, self.cells[-1].connect_to_target(
-                            None, self.params['threshold']))
-                elif type == 'extinput':
-                    # print('type',type)
-                    # to find param index, take difference between REAL gid
-                    # here and gid start point of the items
-                    p_ind = gid - self.gid_dict['extinput'][0]
-                    # now use the param index in the params and create
-                    # the cell and artificial NetCon
-                    self.extinput_list.append(ExtFeed(
-                        type, None, self.p_ext[p_ind], gid))
-                    pc.cell(
-                        gid, self.extinput_list[-1].connect_to_target(
-                            self.params['threshold']))
-                elif type in self.p_unique.keys():
-                    gid_post = gid - self.gid_dict[type][0]
-                    cell_type = self.gid_to_type(gid_post)
-                    # create dictionary entry, append to list
-                    self.ext_list[type].append(ExtFeed(
-                        type, cell_type, self.p_unique[type], gid))
-                    pc.cell(
-                        gid, self.ext_list[type][-1].connect_to_target(
-                            self.params['threshold']))
+                Cell = type2class[src_type]
+                if src_type in ('L2_pyramidal', 'L5_pyramidal'):
+                    self.cells.append(Cell(gid, src_pos, self.params))
                 else:
-                    print("None of these types in Net()")
-                    exit()
+                    self.cells.append(Cell(gid, src_pos))
+
+                pc.cell(gid, self.cells[-1].connect_to_target(
+                        None, self.params['threshold']))
+
+            # external inputs are special types of pseudo-cells
+            # 'common': all cells impacted with identical TIMING of spike
+            # events. NB: cell types can still have different weights for how
+            # such 'common' spikes influence them
+            elif src_type == 'common':
+                # print('cell_type',cell_type)
+                # to find param index, take difference between REAL gid
+                # here and gid start point of the items
+                p_ind = gid - self.gid_dict['common'][0]
+
+                # new ExtFeed: target cell type irrelevant (None) since input
+                # timing will be identical for all cells
+                # XXX common_feeds is a list of dict (comp. unique_feeds)
```

Comment:
what's comp. ? compare against? I would just delete that part in the brackets for now. We'll need to clarify it somewhere more prominent.